docs/examples/compute/instance/compute.tf

```diff
@@ -1,15 +1,16 @@
 resource "oci_core_instance" "TFInstance" {
+  count = "${var.NumInstances}"
   availability_domain = "${lookup(data.oci_identity_availability_domains.ADs.availability_domains[var.AD - 1],"name")}"
   compartment_id = "${var.compartment_ocid}"
-  display_name = "TFInstance"
+  display_name = "TFInstance${count.index}"
```

Comment:
Here, you're changing the main compute instance example from creating a single instance to n-instances. This is a general question: do we want a complex, mega-example that exercises a plethora of optional configurations (one compute example that does _everything_ you can do with an instance), or do we want targeted examples with overlap (single instance, n-instances, with metadata, with ipxeScript, etc.)? If we go with the latter, then you should leave this example as is & create a new one next to it.

I personally prefer multiple targeted examples with an overlap. They are scenario-based, easier to understand & target the different requirements & complexities the different users have. Plus, we already have a mega-example that exercises all the options (albeit, not runnable) in the docs.

Reply:
I thought about this as well, and I agree that we should have multiple targeted examples with overlap (which is the point of this example). Really complex mega samples belong in the solutions folder.

The reason I decided to replace this particular example rather than creating a new one, is because the old example was pretty trivial and would've repeated 90% of the syntax of this example (i.e. this sample is a superset of what the old sample tried to demonstrate).

Reply:
A create _n_ _x_ example is very much needed, this is a good place for it. I'd like to see most configurations use this model, even if _n_ = 1. The SQL Server Always On configuration uses this but it's a complicated configuration overall, that bit is hard to tease out.